fabfile.py

```diff
@@ -0,0 +1,264 @@
+
+
+def list_aws_instances(verbose=False, state='all'):
+    conn = get_ec2_connection()
+
+    reservations = conn.get_all_reservations()
+    instances = []
+    for res in reservations:
+        for instance in res.instances:
+            if state == 'all' or instance.state == state:
+                instance = {
+                    'id': instance.id,
+                    'type': instance.instance_type,
+                    'image': instance.image_id,
+                    'state': instance.state,
+                    'instance': instance,
+                }
+                instances.append(instance)
+    env.instances = instances
+    if verbose:
+        import pprint
+        pprint.pprint(env.instances)
+
+
+def select_instance(state='running'):
+    if env.get('active_instance', False):
+        return
+
+    list_aws_instances(state=state)
+
+    prompt_text = "Please select from the following instances:\n"
+    instance_template = " %(ct)d: %(state)s instance %(id)s\n"
+    for idx, instance in enumerate(env.instances):
+        ct = idx + 1
+        args = {'ct': ct}
+        args.update(instance)
+        prompt_text += instance_template % args
+    prompt_text += "Choose an instance: "
+
+    def validation(input):
+        choice = int(input)
+        if choice not in range(1, len(env.instances) + 1):
+            raise ValueError("%d is not a valid instance" % choice)
+        return choice
+
+    choice = prompt(prompt_text, validate=validation)
+    env.active_instance = env.instances[choice - 1]['instance']
+
+
+def run_command_on_selected_server(command):
+    select_instance()
+    selected_hosts = [
+        'ubuntu@' + env.active_instance.public_dns_name
+    ]
+    execute(command, hosts=selected_hosts)
+
+
+def _setup_suite():
+    sudo('apt-get update')
+    if files.exists('/tmp/supervisor.sock'):
+        sudo('unlink /tmp/supervisor.sock')
+    if files.exists('/var/run/supervisor.sock'):
+        sudo('unlink /var/run/supervisor.sock')
+    sudo('apt-get install -y '
+         'nginx git python-pip postgresql '
+         'postgresql-contrib libpq-dev python-dev')
+    if not files.exists(
+        'etc/nginx/sites-available/original-default',
+        use_sudo=True
+    ):
+        sudo(
+            'cp /etc/nginx/sites-available/default '
+            '/etc/nginx/sites-available/original-default'
+        )
+
+    put(local_path="~/projects/t-buddies/simple_nginx_conf",
+        remote_path="/etc/nginx/sites-available/default",
+        use_sudo=True)
+
+    append('/etc/nginx/sites-available/default',
+           "server {listen 80;server_name " +
+           env.active_instance.public_dns_name + "/;"
+           "access_log /var/log/nginx/test.log;location /"
+           "{proxy_pass http://127.0.0.1:8080;proxy_set_header Host $host;"
+           "proxy_set_header X-Real-IP $remote_addr;"
+           "proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;"
+           "}}")
```

Comment:
There are some nice templating commands you could use that would be a bit easier to deal with than writing the nignx conf file line by line.